interpreter/spec/values.ml

```diff
@@ -3,10 +3,48 @@ open Types
 
 (* Values and operators *)
 
-type ('i32, 'i64, 'f32, 'f64) op =
-  I32 of 'i32 | I64 of 'i64 | F32 of 'f32 | F64 of 'f64
+type ('i32, 'i64, 'f32, 'f64, 'obj) op =
+  I32 of 'i32 | I64 of 'i64 | F32 of 'f32 | F64 of 'f64 | Obj of obj_type
+and value = (I32.t, I64.t, F32.t, F64.t, obj_type) op
+and obj_type = {
+    type_index : int32;
+    fields : value array option;
```

Comment:
There shouldn't be any uninitialized fields in the semantics, so you should be able to get by without an option here.